tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,364 @@
+import datetime
+import logging
+from unittest import mock
+
+import pytest
+from h_matchers import Any
+
+from h.models import Job
+from h.search.client import Client
+from h.search.index import BatchIndexer
+from h.services.search_index._queue import Queue
+
+
+class TestAddSyncAnnotationJob:
+    def test_it(self, db_session, queue):
+        queue.add("test_annotation_id", "test_tag", scheduled_at=one_week_from_now)
+
+        assert db_session.query(Job).all() == [
+            Any.instance_of(Job).with_attrs(
+                dict(
+                    enqueued_at=Any.instance_of(datetime.datetime),
+                    scheduled_at=one_week_from_now,
+                    tag="test_tag",
+                    kwargs={"annotation_id": "test_annotation_id"},
+                )
+            ),
+        ]
+
+    def test_scheduled_at_defaults_to_now(self, db_session, queue):
+        queue.add("test_annotation_id", "test_tag")
+
+        assert db_session.query(Job).all() == [
+            Any.instance_of(Job).with_attrs(
+                dict(
+                    enqueued_at=Any(),
+                    scheduled_at=Any.instance_of(datetime.datetime),
+                    tag=Any(),
+                    kwargs=Any(),
+                )
+            ),
+        ]
+
+
+class TestSyncAnnotations:
+    def test_it_does_nothing_if_the_queue_is_empty(self, batch_indexer, caplog, queue):
+        queue.sync()
+
+        batch_indexer.index.assert_not_called()
+        assert caplog.records == []
+
+    def test_it_ignores_jobs_that_arent_scheduled_yet(
+        self, annotation_ids, batch_indexer, caplog, queue
+    ):
+        queue.add_all(
+            annotation_ids,
+            tag="test",
+            scheduled_at=one_week_from_now,
```

Comment:
If you knock the trailing commas off of all these statements and the signatures, and format again significant vertical space can be reclaimed in this file (about 75 lines)

Reply:
:+1: Done